stdlib/public/core/DictionaryCasting.swift

```diff
@@ -66,7 +72,24 @@ public func _dictionaryDownCast<BaseKey, BaseValue, DerivedKey, DerivedValue>(
       _immutableCocoaDictionary: source._variant.asNative.bridged())
   }
 #endif
-  return _dictionaryDownCastConditional(source)!
+
+  // Note: We can't delegate this call to _dictionaryDownCastConditional,
```

Comment:
This general pattern gets repeated quite a bit, is there enough commonality here that it could be factored out into its own function?

Reply:
That's a good idea! A closure-based method could unify all three casting flavors. A slight wrinkle is that it'll have to allow for the early exit in the optional downcast, but that's fine.